### PR TITLE
SearchKit - Support implied operators in exposed search forms

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This opens up a lot more possibilities for exposed filters with searchKit displays.

Before
----------------------------------------
Only single-value filters accepted.

After
----------------------------------------
New "Multi-Select" and "Search by range" options will allow searches like `IN(...)` or `BETWEEN [...]` respectively.

image

The most complex one is Date fields, which allows you to change the widget to Select *and* enable "Search by range". If you do both then you get the familiar dropdown of relative dates plus a "Choose Date Range" option:

image

Technical Details
----------------------------------------
Adds a basic unit test to ensure operators are functioning.

Comments
----------------------------------------
This doesn't implement exposed filters, but it doesn't box us out of that possibility either. The filter syntax will accept operators.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@colemanw style issue

File: ext/afform/admin/ang/afGuiEditor/elements/afGuiField.component.js [hunk 1]
```diff
@@ -30,14 +36,35 @@
...
       this.getDefn = function() {
         var defn = afGui.getField(ctrl.container.getFieldEntityType(ctrl.node.name), ctrl.node.name);
-        return defn ||  {
+        defn = defn || {
           label: ts('Untitled'),
-          requred: false,
-          input_attrs: []
+          required: false
         };
+        defn.input_attrs = _.isEmpty(defn.input_attrs) ? {} : defn.input_attrs;
```
Comment (seamuslee001) [inline ext/afform/admin/ang/afGuiEditor/elements/afGuiField.component.js:66]: are we switching from an empty array to an empty object here?

Comment (colemanw) [reply ext/afform/admin/ang/afGuiEditor/elements/afGuiField.component.js:66]: Yes the former was a mistake made by my php-centric brain.

**Comment by eileenmcnaughton** [comment]
I gave this a quick spin & it seems to work fine although I note that 

1) the date range I selected doesn't seem to show once selected - here I have 'this calendar month selected' on the test site here http://core-19959-95lp4.test-1.civicrm.org:8001/civicrm/post

image

2) the cache clearing is really tricky - when you save the form it does not clear the cache out so if you add a field to the form it doesn't show up until you take a separate cache-clearing action

3) I was able to select 'search by range' for 'prefix_id' which seems a bit odd but I suppose it permits people to do the sorts of weird and wonderful things people always seem to come up with

4) I wasn't able to do search by range on postal code - which makes sense from a metadata POV as it is a string but we have handling on advanced search to permit it

@colemanw I don't consider the above to be blocking although the first is perhaps 'near blocking'. I'm happy to merge & leave those things out of scope if you want

**Comment by colemanw** [comment]
@eileenmcnaughton I've fixed `#1` and `#4`. I think `#3` is fine.
I reproduced `#2` by enabling asset caching & if I can find the cause I'll put up a separate PR for that one.
This should be good to merge.

**Comment by eileenmcnaughton** [comment]
I've retested & 1 is fixed

3 is quasi fixed - see how 900 is included here - I don't think addressing that needs to be in scope so I'm merging

image

**Pull request merged**